packages/core/src/telemetry/activity-detector.test.ts

```diff
@@ -0,0 +1,193 @@
+
+      detector.recordActivity();
+      vi.advanceTimersByTime(500);
+
+      const timeSince = detector.getTimeSinceLastActivity();
+      expect(timeSince).toBe(500);
+
+      vi.useRealTimers();
+    });
+  });
+
+  describe('getLastActivityTime', () => {
+    it('should return the timestamp of last activity', () => {
+      const before = Date.now();
+      detector.recordActivity();
+      const activityTime = detector.getLastActivityTime();
+      const after = Date.now();
+
+      expect(activityTime).toBeGreaterThanOrEqual(before);
+      expect(activityTime).toBeLessThanOrEqual(after);
+    });
+  });
+});
+
+describe('Global Activity Detector Functions', () => {
+  beforeEach(() => {
+    // Reset global instance
+    resetGlobalActivityDetector();
+  });
+
+  describe('initializeActivityDetector', () => {
+    it('should create and return a global instance', () => {
+      const detector = initializeActivityDetector();
+      expect(detector).toBeInstanceOf(ActivityDetector);
+    });
+
+    it('should return same instance on multiple calls', () => {
+      const detector1 = initializeActivityDetector();
+      const detector2 = initializeActivityDetector();
+      expect(detector1).toBe(detector2);
+    });
+
+    it('should accept custom idle threshold', () => {
+      const detector = initializeActivityDetector(5000);
+      expect(detector).toBeInstanceOf(ActivityDetector);
+    });
+  });
+
+  describe('getActivityDetector', () => {
+    it('should return null when not initialized', () => {
+      expect(getActivityDetector()).toBeNull();
+    });
+
+    it('should return initialized instance', () => {
+      const detector = initializeActivityDetector();
+      expect(getActivityDetector()).toBe(detector);
+    });
+  });
+
+  describe('recordUserActivity', () => {
+    it('should not initialize detector when none exists', () => {
+      expect(getActivityDetector()).toBeNull();
+
+      expect(() => recordUserActivity()).not.toThrow();
+
+      // Detector remains uninitialized; call is a no-op
+      expect(getActivityDetector()).toBeNull();
+    });
+
+    it('should record activity on existing detector', () => {
+      const detector = initializeActivityDetector();
+      const beforeTime = detector.getLastActivityTime();
+
+      vi.useFakeTimers();
+      vi.advanceTimersByTime(100);
+
+      recordUserActivity();
+
+      const afterTime = detector.getLastActivityTime();
+      expect(afterTime).toBeGreaterThan(beforeTime);
+
+      vi.useRealTimers();
+    });
+  });
+
+  describe('isUserActive', () => {
+    it('should return false when no detector exists', () => {
+      expect(isUserActive()).toBe(false);
+    });
+
+    it('should return detector state when exists', () => {
+      initializeActivityDetector(1000);
+      expect(isUserActive()).toBe(true);
+
+      vi.useFakeTimers();
+      vi.advanceTimersByTime(2000);
+
+      expect(isUserActive()).toBe(false);
+
+      vi.useRealTimers();
```

Comment:
calling useRealTimers and useFakeTimers should be done in
beforeEach
and 
afterEach blocks rather than integrated into individual tests.